Suricata6: initial commit

Grommish opened the pull request:
Signed-off-by: Donald Hoskins <grommish@gmail.com>

Maintainer: me
Compile tested: mips64 mipsel / main branch 0f7a3288e10fb244a33d80230e7b04b323668821

Description:
Requires Rust/Cargo to compile (https://github.com/openwrt/packages/pull/13916)

jneuhauser commented on net/suricata6/Makefile:
```diff
@@ -0,0 +1,127 @@
+<<<<<<< HEAD
+=======
+	--enable-ebpf-build \
+>>>>>>> 56cb5fc21... Suricata6: initial commit
```
You shoudl fix that merge conflict!

Grommish replied:
I didn't have that on my end.. i wonder what happened.. In any case, I had to update ;) Thank you!

BKPepe commented on net/suricata6/Makefile:
```diff
@@ -0,0 +1,112 @@
+PKG_SOURCE_DATE:=2020-10-08
+PKG_SOURCE_VERSION:=95729e923f39b5adda31e4843a1b6ee06edbb0c8
+PKG_SOURCE_URL:=https://github.com/OISF/suricata.git
+PKG_HASH:=skip
```
Why? Please don't do this. If you are not sure which value put there, take a look at documentation or the build system can add the value for you.

Documentation here: https://openwrt.org/docs/guide-developer/packages#use_source_repository

Grommish replied:
I am skipping the HASH because it's not needed for testing.  This will be completed with a FIXUP for a calc'ed SHA256 hash for the finalized .tar.xz file before I release it

BKPepe replied:
I think this is requirement otherwise Github Actions does not pass for this PR.

BKPepe commented on net/suricata6/Makefile:
```diff
@@ -0,0 +1,112 @@
+#
+# Copyright (C) 2006-2015 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
```
Omit or assign it to your own.
More details: https://github.com/openwrt/packages/blame/master/CONTRIBUTING.md#L45

BKPepe commented on net/suricata6/Makefile:
```diff
@@ -0,0 +1,112 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=suricata
+PKG_VERSION:=6.0.0-beta1
```
There is released version 6.0.1 and you dont need to use beta version anymore.

Grommish replied:
And they are working on 7.  I've not updated this since 6.0.1 went live

BKPepe commented on net/suricata6/Makefile:
```diff
@@ -0,0 +1,112 @@
+PKG_NAME:=suricata
+PKG_VERSION:=6.0.0-beta1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_DATE:=2020-10-08
+PKG_SOURCE_VERSION:=95729e923f39b5adda31e4843a1b6ee06edbb0c8
+PKG_SOURCE_URL:=https://github.com/OISF/suricata.git
```
You can use codeload.

BKPepe commented on net/suricata6/Makefile:
```diff
@@ -0,0 +1,112 @@
+PKG_SOURCE_URL:=https://github.com/OISF/suricata.git
+PKG_HASH:=skip
+
+SURICATA-UPDATE_SOURCE_DATE:=2020-10-05
+SURICATA-UPDATE_SOURCE_VERSION:=86e530cbbfc78ac327072750a12edf888e6b8cae
+SURICATA-UPDATE_SOURCE_URL:=https://github.com/OISF/suricata-update.git
+SURICATA-UPDATE_HASH:=skip
```
Why do you need this? You are not using this anywhere.

BKPepe commented on net/suricata6/Makefile:
```diff
@@ -0,0 +1,112 @@
+	--build=$(RUSTC_HOST_ARCH) \
+	--with-suricata-update
+
+define Build/InstallDev
+	$(call HostPython3/Run,,$(HOST_PYTHON3_PIP_INSTALL_HOST) PyYAML)
+endef
```
Why are you doing this? You can not use this https://github.com/openwrt/packages/tree/master/lang/python#host-side-python-packages-for-build if you want have host package pyyaml? Why do you want pyyaml at all installed as host?

BKPepe commented on net/suricata6/Makefile:
```diff
@@ -0,0 +1,112 @@
+
+define Build/Prepare
+	$(call Build/Prepare/Default)
+	cd $(PKG_BUILD_DIR) && git clone https://github.com/OISF/libhtp.git
```
Wouldn't it be better to package libhtp separately?
Something like this: https://gitlab.nic.cz/turris/turris-os-packages/-/blob/master/libs/libhtp/Makefile

Grommish replied:
I thought about it, but then I've got to do ANOTHER PR just for LibHTP and wait for whoever to approve it.  I've already got a chain of contingencies here (Rust for Suricata for Suricata LibHTP and Update?) and didn't want to string them out to much.  I certainly don't MIND creating a libHTP - but since libHTP is used in the Suricata build directory on compile, I'll have to see about doing it that way if it's an issue :)

Grommish commented:
Updated for issues pertaining to libpcap and tyepdefs.

Built and tested on mipsel_24kc (ER10x)

Grommish commented:
Updated for deps and Config.in changes.  Also, added https://github.com/openwrt/packages/pull/15287 to address using a separate LibHTP

Grommish commented:
Updated Makefile, added `x86_64` logic to avoid `-fPIE` calls.

Identified an issue with `suricata6` on at least the `Mips64` platform (https://forum.suricata.io/t/allocating-host-hash-failed-max-host-memcap-is-smaller-than-projected-hash-size/2298) - other platforms may be effected under OpenWrt (untested).

predators46 commented:
@Grommish 

i tested on openwrt 22.03 got my suricata6 having error
`    Updating crates.io index
error: process didn't exit successfully: `rustc -vV` (exit status: 127)                                 --- stderr
rustc: error while loading shared libraries: libLLVM-14.so: cannot open shared object file: No such file or directory`


I see this file exists `: libLLVM-14.so`

Grommish commented:
> @Grommish
> 
> i tested on openwrt 22.03 got my suricata6 having error ` Updating crates.io index error: process didn't exit successfully:`rustc -vV` (exit status: 127) --- stderr rustc: error while loading shared libraries: libLLVM-14.so: cannot open shared object file: No such file or directory`
> 
> I see this file exists `: libLLVM-14.so`

This happened because LLVM was split off.  Once I get done making changes to `rust-lang` and `llvm` and finalize them, I'll update the other PRs as appropriate.

For now, you can change the suricata `Makefile` to add the following:
```
LLVM_DIR:=$(STAGING_DIR_HOST)/llvm-rust
LD_LIBRARY_PATH:=$(LLVM_DIR)/lib
MAKE_FLAGS += $(LD_LIBRARY_PATH)
```

That should allow you to build out with the standard `make` call suricata uses and point it to the right place.

tiagogaspar8 commented:
Hi everyone,
Now that Rust support is merged will this be moving forward?
In case any help is needed just say.
Thanks!

tiagogaspar8 commented:
Hi @Grommish ,

Would you be open to working on this?
If not, would it be OK to take your work and try it myself?

trippleflux commented:
@tiagogaspar8 
Afaik, @Grommish is not active for quite long time (~1-2 years?), last time I have had read if I am not mistaken he was having problems with his PC. I vote for continuing his old suricata works for suricata 7 with your expertise.

tiagogaspar8 commented:
I'm currently working on it, but I'm having some issues, I have no idea how the rust host build works... but I'm trying to figure it out, let's see what I can do.
If anyone wants to help this rust noob just say somthing :smile:

BKPepe commented:
Closing, OP is inactive.
